packages/data/src/batcher.ts

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ReadStream } from 'fs';
+import { Connection, Logger, Messages, SfdxError } from '@salesforce/core';
+import parse = require('csv-parse');
+import { Batch, BatchInfo, BatchResultInfo, JobInfo } from 'jsforce';
+import { UX } from '@salesforce/command';
+import { Job as jsforceJob } from 'jsforce/job';
+
+const BATCH_RECORDS_LIMIT = 10000;
+const POLL_FREQUENCY_MS = 5000;
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/data', 'batcher');
+
+export type Batches = Array<Array<Record<string, string>>>;
+export type Job = jsforceJob & { id: string };
+
+export type BulkResult = {
+  $: {
+    xmlns: string;
+  };
+  id: string;
+  jobId: string;
+  state: string;
+  createdDate: string;
+  systemModStamp: string;
+  numberRecordsProcessed: string;
+  numberRecordsFailed: string;
+  totalProcessingTime: string;
+  apiActiveProcessingTime: string;
+  apexProcessingTime: string;
+};
+
+export class Batcher {
+  /**
+   * get and display the job status; close the job if completed
+   *
+   * @param conn {Connection}
+   * @param jobId {string}
+   * @param ux
+   * @param doneCallback
+   */
+  public static async fetchAndDisplayJobStatus(
+    conn: Connection,
+    jobId: string,
+    ux: UX,
+    doneCallback?: (...args: [{ job: JobInfo }]) => void
+  ): Promise<JobInfo> {
+    const job: jsforceJob = conn.bulk.job(jobId);
+    const jobInfo: JobInfo = await job.check();
+
+    Batcher.bulkBatchStatus(jobInfo, ux);
+
+    if (doneCallback) {
+      doneCallback({ job: jobInfo });
+    }
+
+    return jobInfo;
+  }
+
+  public static bulkBatchStatus(
+    summary: JobInfo | BatchInfo,
+    ux: UX,
+    results?: BatchResultInfo[],
+    batchNum?: number,
+    isJob?: boolean
+  ): void {
+    ux.log('');
```

Comment:
Is there any way to avoid passing in `ux`? Again, thinking about this from a usability perspective, people may not be using this library in the context of a plugin so they wouldn't have and wouldn't need the `ux` class